sql-ballerina/Module.md

```diff
@@ -0,0 +1,302 @@
+property in the record). Next, the SELECT query is executed via the `query` remote function of the client by passing that 
+result set type. Once the query is executed, each data record can be retrieved by looping the result set. The `stream` 
+returned by the select operation holds a pointer to the actual data in the database and it loads data from the table 
+only when it is accessed. This stream can be iterated only once. 
+
+```ballerina
+// Define a type to represent the results.
+type Student record {
+    int id;
+    int age;
+    string name;
+};
+
+// Select the data from the database table. The query parameters are passed 
+// directly. Similar to the `execute` examples, parameters can be passed as
+// sub types of `sql:TypedValue` as well.
+int id = 10;
+int age = 12;
+sql:ParameterizedQuery query = `SELECT * FROM students
+                                WHERE id < ${id} AND age > ${age}`;
+stream<Student, sql:Error> resultStream = 
+        <stream<Student, sql:Error>> dbClient->query(query, Student);
+
+// Iterating the returned table.
+error? e = resultStream.forEach(function(Student student) {
+   //Can perform any operations using 'student' and can access any fields in the returned record of type Student.
+});
+```
+
+Defining the return type is optional and you can query the database without providing the result type. Hence, 
+the above example can be modified as follows with an open record type as the return type. The property name in the open record 
+type will be the same as how the column is defined in the database. 
+
+```ballerina
+// Select the data from the database table. The query parameters are passed 
+// directly. Similar to the `execute` examples, parameters can be passed as 
+// sub types of `sql:TypedValue` as well.
+int id = 10;
+int age = 12;
+sql:ParameterizedQuery query = `SELECT * FROM students
+                                WHERE id < ${id} AND age > ${age}`;
+stream<record{}, sql:Error> resultStream = dbClient->query(query);
+
+// Iterating the returned table.
+error? e = resultStream.forEach(function(record{} student) {
+    //Can perform any operations using 'student' and can access any fields in the returned record.
+});
+```
+
+There are situations in which you may not want to iterate through the database and in that case, you may decide
+to only use the `next()` operation in the result `stream` and retrieve the first record. In such cases, the returned
+result stream will not be closed, and you have to explicitly invoke the `close` operation on the 
+`sql:Client` to release the connection resources and avoid a connection leak as shown below.
+
+```ballerina
+stream<record{}, sql:Error> resultStream = 
+            dbClient->query("SELECT count(*) as total FROM students");
+
+record {|record {} value;|}|error? result = resultStream.next();
+
+if result is record {|record {} value;|} {
+    //valid result is returned.
```

Comment:
```suggestion
    //A valid result is returned.
```

Reply:
Change Added in https://github.com/ballerina-platform/module-ballerina-sql/pull/228/commits/2656f12c62276e15a256c133465cb60304d0b041